samples/mixed/consumer/src/main.rs

```diff
@@ -0,0 +1,242 @@
+    debug!("Received the response for the find_by_id request. The DTDL is:\n{dtdl}");
+
+    debug!("Parsing the DTDL.");
+    let mut parser = ModelParser::new();
+    let json_texts = vec![dtdl];
+    let model_dict_result = parser.parse(&json_texts);
+    if let Err(error) = model_dict_result {
+        return Err(format!("Failed to parse the DTDL: {error}"));
+    }
+    let model_dict = model_dict_result.unwrap();
+    debug!("The DTDL parser has successfully parsed the DTDL.");
+
+    // Create the id (as a DTMI) for the send_notification command.
+    let dtmi_id: Option<Dtmi> = create_dtmi(entity_id);
+    if dtmi_id.is_none() {
+        return Err(String::from("Unable to create the dtmi"));
+    }
+
+    // Get the entity from the DTDL for the dtmi id.
+    let entity_result = model_dict.get(&dtmi_id.unwrap());
+    if entity_result.is_none() {
+        return Err(String::from("Unable to find the entity"));
+    }
+    let entity = entity_result.unwrap();
+
+    // Get the URI property from the entity.
+    let uri_property_result = entity.undefined_properties().get(sdv::property::uri::ID);
+    if uri_property_result.is_none() {
+        return Err(String::from("Unable to find the URI property"));
+    }
+    let uri_property = uri_property_result.unwrap();
+
+    // Get the value for the URI property.
+    let uri_property_value_result = uri_property.get("@value");
+    if uri_property_value_result.is_none() {
+        return Err(String::from("Unable to find the value for the URI."));
+    }
+    let uri_property_value = uri_property_value_result.unwrap();
+    let uri_str_option = uri_property_value.as_str();
+    let provider_uri = String::from(uri_str_option.unwrap());
+    info!("The provider URI for entity {entity_id} is {provider_uri}");
+
+    Ok(provider_uri)
+}
+
+async fn send_subscribe_request(
+    provider_uri: &str,
+    entity_id: &str,
+    consumer_uri: &str,
+) -> Result<(), Box<dyn std::error::Error>> {
+    debug!("Sending a subscribe request for {entity_id}.");
+    let mut client = ProviderClient::connect(provider_uri.to_string()).await?;
+    let request = tonic::Request::new(SubscribeRequest {
+        entity_id: String::from(entity_id),
+        consumer_uri: consumer_uri.to_string(), // Devskim: ignore DS137138
+    });
+    let _response = client.subscribe(request).await?;
+
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Setup logging.
+    Builder::new().filter(None, LevelFilter::Info).target(Target::Stdout).init();
+
+    info!("The Consumer has started.");
+
+    // Setup the HTTP server.
+    let consumer_authority = String::from("[::1]:60010");
+    let addr: SocketAddr = consumer_authority.parse()?;
+    let consumer_impl = consumer_impl::ConsumerImpl::default();
+    let server_future =
+        Server::builder().add_service(ConsumerServer::new(consumer_impl)).serve(addr);
+
+    let show_notification_command_provider_uri =
```

Comment:
If you want the error to be the panic message, you can do 

```
let show_notification_command_provider_uri =
       get_provider_uri(sdv::vehicle::cabin::infotainment::hmi::show_notification::ID).await.unwrap();
```

unwrap() will panic if it's an error and will output the error as the message.

Reply:
Fixed.